### PR TITLE
Use deezer for fetching artist images

### PR DESCRIPTION
**Pull request by paolovalerdi**
Self descriptive

### CODE REVIEW

**Comment by AdrienPoupa** [comment]
Oh, awesome! I was just looking at Discogs and Spotify APIs but since Deezer does not require authentication, that's better

**Comment by paolovalerdi** [comment]
Do not merge this yet. I'm seeing this crash: https://del.dog/akiqigezih.sql

**Comment by AdrienPoupa** [comment]
OK, let me know when you're done

**Comment by paolovalerdi** [comment]
Are you able to reproduce this issue? I haven´t been able to

File: app/src/main/java/com/poupa/vinylmusicplayer/glide/artistimage/ArtistImageFetcher.java [hunk 1]
```diff
@@ -58,22 +58,25 @@ public DataSource getDataSource() {
...
+                        DeezerResponse lastFmArtist = response.body();
+                        Log.d("DEEZER", String.valueOf(lastFmArtist));
+                        if (lastFmArtist == null) {
                             callback.onLoadFailed(new Exception("No artist image url found"));
                             return;
                         }
 
-                        String url = LastFMUtil.getLargestArtistImageUrl(lastFmArtist.getArtist().getImage());
+                        String url = lastFmArtist.getData().get(0).getPictureMedium();
+                        Log.d("DEEZER", url);
```
Comment (AdrienPoupa) [inline app/src/main/java/com/poupa/vinylmusicplayer/glide/artistimage/ArtistImageFetcher.java]: Can you remove the log lines?

**Comment by AdrienPoupa** [comment]
Yes I can reproduce it. The app crashes at start
```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.poupa.vinylmusicplayer.debug, PID: 13802
    java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
        at java.util.ArrayList.get(ArrayList.java:437)
        at com.poupa.vinylmusicplayer.glide.artistimage.ArtistImageFetcher$1.onResponse(ArtistImageFetcher.java:78)
```

File: app/src/main/java/com/poupa/vinylmusicplayer/glide/artistimage/ArtistImageFetcher.java (hunk 1 above)
Comment (paolovalerdi) [reply app/src/main/java/com/poupa/vinylmusicplayer/glide/artistimage/ArtistImageFetcher.java]: Oh yes, I will I forgot about them

**Comment by paolovalerdi** [comment]
Weird, I'll check on my device (Essential phone) and emulator running O there's no crash at all

**Comment by AdrienPoupa** [comment]
Basically you call `.get(0)` on a list that could be empty...

image

**Comment by paolovalerdi** [comment]
Yes lmao, just noticed that thanks.

**Comment by AdrienPoupa** [comment]
```
                        DeezerResponse deezerResponse = response.body();
                        if (deezerResponse == null || deezerResponse.getData().size() == 0) {
                            callback.onLoadFailed(new Exception("No artist image url found"));
                            return;
                        }

                        String url = deezerResponse.getData().get(0).getPictureMedium();
                        if (TextUtils.isEmpty(url) || TextUtils.isEmpty(url.trim())) {
                            callback.onLoadFailed(new Exception("No artist image url found"));
                            return;
                        }
```

works fine, so if you can just rename the lastFm variable and get rid of the log instructions, I can go ahead and merge it :)

**Comment by paolovalerdi** [comment]
Done

**Pull request merged**